plugin/src/main/java/com/playmonumenta/redissync/DataEventListener.java

```diff
@@ -523,9 +526,35 @@ public void playerDataLoadEvent(PlayerDataLoadEvent event) {
 			event.setData(nbtTagCompound);
 
 			mLogger.fine(() -> "Processing PlayerDataLoadEvent took " + (System.currentTimeMillis() - startTime) + " milliseconds on main thread");
-		} catch (IOException | InterruptedException | ExecutionException ex) {
-			mLogger.severe("Failed to load player data: " + ex);
-			ex.printStackTrace();
+		} catch (Throwable ex) {
+			mLogger.severe("!!! Failed to load player data !!!");
+			mLogger.log(Level.SEVERE, "Exception", ex);
+
+			final var rootPath = MonumentaRedisSync.getInstance().getDataFolder().toPath()
+				.resolve("data-fail-report-%s-%s-%s".formatted(
+					System.currentTimeMillis(),
+					player.getName(),
+					player.getUniqueId()
+				));
+
+			mLogger.severe("Writing data files to for analysis..." + rootPath);
+
+			try {
+				Files.createDirectories(rootPath);
+				Files.write(rootPath.resolve("data.nbt"), dataFuture.get());
```

Comment:
We might want to handle if these futures throw an exception, most likely due to a failed connection to Redis. Any other exception would be even more concerning.

Also consider testing this by intentionally throwing an exception, just to make sure it writes as expected. It probably will, though.